CMakeLists.txt

```diff
@@ -4,27 +4,43 @@
 
 cmake_minimum_required(VERSION 3.15)
 
+# Skip generating proto files with Protobuf compiler and use prebuilt files
+# Will still build Protobuf as libprotobuf is required for linking
+option(WSL2_CROSS_COMPILE "Building on WSL2 for Windows" OFF)
+
+# Only valid OS configuration should be in WSL2 context
+# Should not work wtih WIN32, APPLE, etc.
+if (WSL2_CROSS_COMPILE AND UNIX AND NOT APPLE)
+    set(CMAKE_TOOLCHAIN_FILE cmake/toolchain/x86-64-w64-mingw32.cmake)
+elseif (WSL2_CROSS_COMPILE)
+    message(FATAL_ERROR "Invalid environment.\
+                        WSL2_CROSS_COMPILE=ON for WSL only")
+endif ()
+
 project(BlankBot)
 
-# Specify output directories.
-set(EXECUTABLE_OUTPUT_PATH "${PROJECT_BINARY_DIR}/bin")
+include(FetchContent)
 
-# Use statically linked runtime
-set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+option(BUILD_FOR_LADDER "Create build for the AIArena ladder" OFF)
 
 # Build with c++14 support, required by sc2api.
 set(CMAKE_CXX_STANDARD 14)
 
-# Dont build examples in the sc2api submodule.
-set(BUILD_API_EXAMPLES OFF CACHE INTERNAL "" FORCE)
+# Specify output directories.
+set(EXECUTABLE_OUTPUT_PATH "${PROJECT_BINARY_DIR}/bin")
 
-# Dont build tests in the sc2api submodule.
-set(BUILD_API_TESTS OFF CACHE INTERNAL "" FORCE)
+if (MSVC)
+    # Setup MSVC parallelized builds
```

Comment:
Style: Please, finish the comments with dots for consistency (or drop dots everywhere).
Here and in the other places.